### ISSUE TITLE
'AioServer.shutdown' was never awaited for kserve fastapi server

### ISSUE THREAD

**Issue by navneetraju** [kserve/kserve#2694]
/kind bug

**What steps did you take and what happened:**
Create a simple kserve model:
```
import kserve
from typing import Dict
import argparse
import logging

logger = logging.getLogger(__name__)

class CustomModel(kserve.Model):
    def __init__(self, name: str):
        super().__init__(name)
        self.name = name
        self.load()

    def load(self):
        self.ready = True

    def predict(self,request: Dict, headers: Dict ) -> Dict:
        logger.info(request)
        return request


DEFAULT_MODEL_NAME = "custom-model"

parser = argparse.ArgumentParser(parents=[kserve.model_server.parser])
parser.add_argument('--model_name', default=DEFAULT_MODEL_NAME,
                    help='The name that the model is served under.')

args, _ = parser.parse_known_args()
if __name__ == "__main__":
    model = CustomModel(name=args.model_name)
    kserve.ModelServer().start([model])
```
And when you run the server:
```
RuntimeError: Event loop is closed
sys:1: RuntimeWarning: coroutine 'AioServer.shutdown' was never awaited
```

**Anything else you would like to add:**
What worked for me is adding this line before initializing the model:
```
multiprocessing.set_start_method('fork')
```

The above issue only happens on mac, and does not seem to happen with linux containers, this is probably something to do with how to OS defaults to creating new processes.

**Comment by yuzisun** [comment]
Since python 3.8 the default multiprocessing start method changed from fork to spawn for Mac and Windows, we are looking to fix this in the patch release 0.10.1.

**Comment by lloydhamilton** [comment]
Hey @yuzisun I am also running into the same issue. Just adding to the thread. The error I am also getting show this:

```
   File "/Users/.../lib/python3.9/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
AttributeError: Can't pickle local object 'request_response.<locals>.app'
```
